src/test/java/com/kodilla/ecommercee/repositoryDao/CartRepositoryTestSuit.java

```diff
@@ -0,0 +1,184 @@
+package com.kodilla.ecommercee.repositoryDao;
+
+import com.kodilla.ecommercee.domain.Cart;
+import com.kodilla.ecommercee.domain.Order;
+import com.kodilla.ecommercee.domain.Product;
+import com.kodilla.ecommercee.domain.Users;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+public class CartRepositoryTestSuit {
+    @Autowired
+    private CartRepository cartRepository;
+
+    private static Long zeroLong = 0L;
+
+    private Cart cart1;
+    private Cart cart2;
+    private Cart cart3;
+
+    private Users user1;
+
+    private Product product1;
+    private Product product2;
+    private Product product3;
+
+    private List<Product> listOfProducts1;
+    private List<Product> listOfProducts2;
+    private List<Product> listOfProducts3;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CartRepositoryTestSuit.class);
+
+    @Before
+    public void beforeTest(){
+        product1 = new Product("Glosniki","czarne",23);
+        product2 = new Product("Telefon","bezprzewodowy",40);
+        product3 = new Product("Pokrowiec","przezroczysty",15);
+
+        listOfProducts1 = new ArrayList<>();
+        listOfProducts2 = new ArrayList<>();
+        listOfProducts3 = new ArrayList<>();
+
+        cart1 = new Cart();
+        cart1.setProductsList(listOfProducts1);
+        cart2 = new Cart();
+        cart2.setProductsList(listOfProducts2);
+        cart3 = new Cart();
+        cart3.setProductsList(listOfProducts3);
+    }
+    @Transactional
+    @Test
+    public void testCartDaoSaveAndFindAll(){
+        //Given
+        cartRepository.save(cart1);
+        cartRepository.save(cart2);
+        cartRepository.save(cart3);
+        //When
+        List<Cart> listOfCarts = cartRepository.findAll();
+        //Then
+        Assert.assertEquals(3,listOfCarts.size());
+        Assert.assertNotEquals(null,cartRepository.findById(cart1.getId()));
```

Comment:
linie 75-77: te asercje powinny mieć w pierwszym argumencie zmienną zeroLong, a w drugim samo cart1.getId()

Reply:
Ola jaki sens jest wyciagac bezposrednio z cart1?? przeciez tutaj sprawdzam czy cart zapisalo sie w rep, wiec Id wyciagam poprzez rep, sprawdzajac Twoja metoda nic nie udowadniam poza tym ze cart1 jako obiekt ma Id

Reply:
Udowadniasz, bo jeśli cart nie zapisałoby się poprawnie w repo to obiekt cart nie miałby id. Zauważ że w konstruktorze cart nie masz pola id - ono się tworzy automatycznie w momencie zapisywania obiektu do bazy danych.

Reply:
Co do nulla, to w zasadzie chyba może być null. Choć oczywiście możesz też wykorzystać stworzoną przez siebie zmienną zeroLong, skoro już ją masz.

Reply:
no wydaje mi sie ze obiekty Tworza sie w postaci Beanow ze springa a do repozytorium zapisujemy komenda save, i tak jak Karol napisal mailem sprawdzic id wyciagajac z bazy danych a nie ze stworzonego zdalnie obiektu

Reply:
nie wiem Ola nie ze sie kloce czy cos tylko przydalby sie komentarz ktoregos z Mentorow

Reply:
Udowadniasz, ponieważ cart1 będzie miał id dopiero wtedy, kiedy zostanie poprawnie zapisany w bazie danych. Zauważ, że konstruktor do cart nie ma pola id - ono przybiera wartość dopiero w momencie zapisania obiektu do bazy danych. Stąd chęć sprawdzenia czy w repozytorium zadziałał "save" sprowadza się do zweryfikowania czy zapisywany obiekt ma id.

Reply:
powtórzyłam komentarz bo myślałam że ten z wcześniej się nie zapisał i nie wysłał, sorki...  Oki, to zrób screena dyskusji i zapytaj mentorów

Reply:
Assert.assertNotEquals(null,cartRepository.findById(cart2.getId())); -> cartRepository.findById zwraca Optional -
Uzyj metody Assert.assertTrue( .... .isPresent() jezeli chcesz sprawdzic czy jest taki rekord w bazie.
Przyklad z GroupRepositoryTestSuit :
Assert.assertTrue(groupFound.isPresent() && groupFound.get().getGroupId().equals(group4Id));

Reply:
@erwinMuz  dzieki, to tak zrobie tylko nie rozumiem czemu nie moze zostac w takiej formie jakiej jest, gdyz dostosowalem sie do tego co Karol mi radzil "sprawdz czy wszystkie relacje sie zapisaly przez wyciaganiete z bazy i sprawdzenie czy !Id= null albo zero" wiec dlatego sie zdziwilem ze to jest zle

Reply:
ok po temacie skasowalem te liniji i tylko sprawdza mi wielkosc listy tak jak np. w zmergowanych testachProduct